frontend/src/components/SignupForm.js

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect, useState } from 'react';
+import { Link } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { signup } from '../reducers/user';
+import { Wrapper, Form } from 'lib';
+
+const SignupForm = () => {
+  const dispatch = useDispatch();
+  const errorMessage = useSelector((store) => store.user.login.errorMessage);
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+
+  // This part can be improved with having one error object as a useState
+  const [nameError, setNameError] = useState(null);
+  const [emailError, setEmailError] = useState(null);
+  const [passwordError, setPasswordError] = useState(null);
+  const [showValidations, setShowValidations] = useState(false);
+  const [formIsValid, setFormIsValid] = useState(false);
+
+  const nameIsValid = () => {
+    if (!name) {
+      setNameError('Name can not be blank');
+      return false;
+    } else if (name.length < 2) {
+      setNameError('Name should be longer than 2 letters');
+      return false;
+    } else if (name.length > 40) {
+      setNameError('Name should be less than 40 letters');
+      return false;
+    }
```

Comment:
It would be nice to implement the error messages that you created in the backend here. 🙂🙂